packages/inventory/src/Stocktake/DetailView/Footer/StocktakeLockButton.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { FC } from 'react';
+import {
+  useTranslation,
+  StocktakeNodeStatus,
+  ToggleButton,
+  useConfirmationModal,
+} from '@openmsupply-client/common';
+
+import { useStocktakeFields } from '../../api';
+
+export const StocktakeLockButton: FC = () => {
+  const t = useTranslation('inventory');
+
+  const { isLocked, status, update } = useStocktakeFields([
+    'isLocked',
+    'status',
+  ]);
+
+  const message = isLocked
+    ? 'messages.unlocked-description'
+    : 'messages.locked-description';
+
+  const getUpdateConfirmation = useConfirmationModal({
+    onConfirm: () => update({ isLocked: !isLocked }),
+    title: t('heading.are-you-sure'),
+    message: t(message),
+  });
```

Comment:
I went with a confirmation because it's an easy way to prevent race conditions of updating other fields, is informative as to what the action does

Reply:
The confirmation is good